packages/pyright-internal/src/parser/stringTokenUtils.ts

```diff
@@ -83,8 +83,40 @@ function completeUnescapedString(incomplete: IncompleteUnescapedString): Unescap
 export function getUnescapedString(stringToken: StringToken): UnescapedString {
     const escapedString = stringToken.escapedValue;
     const isRaw = (stringToken.flags & StringTokenFlags.Raw) !== 0;
+
+    if (isRaw) {
+        return {
+            value: escapedString,
+            unescapeErrors: [],
+            nonAsciiInBytes: false,
+            formatStringSegments: [],
+        };
+    }
+
+    const charCodes: number[] = [];
```

Comment:
maybe there is a minor benefit to pre-allocate the array size. 

something like 
```
const charCodes: number[] = new Array(escapedString.length);
```

Reply:
That won't do quite what you expect:

```
> const arr = new Array(5)
undefined
> arr.push("test")
6
> arr
[ <5 empty items>, 'test' ]
>
```

Array is weird and has "empty items" as a thing. Maybe preallocs are possible by doing the above and setting `arr.length = 0`, but that would require profiling to confirm.